gradle/libs.versions.toml

```diff
@@ -6,20 +6,23 @@ junit = "5.10.2"
 autoservice = "1.1.1"
 
 # otel SDK and API
+# updated from upstream agent with .ci/update-upstream.sh
 opentelemetrySdk = "1.35.0"
 
 # otel protocol (OTLP)
 opentelemetryProto = "1.1.0-alpha"
 
 # otel agent, those two should be updated together
+# updated from upstream agent with .ci/update-upstream.sh
 opentelemetryJavaagent = "2.1.0"
```

Comment:
The opentelemetry-instrumentation bom is transitively imported form the alpha-bom:
Check the bottom of https://repo1.maven.org/maven2/io/opentelemetry/instrumentation/opentelemetry-instrumentation-bom-alpha/2.1.0-alpha/opentelemetry-instrumentation-bom-alpha-2.1.0-alpha.pom

This means we can get rid of explicitly listing and syncing that one too